tests/server/api/types/test_Dataset.py

```diff
@@ -693,3 +733,35 @@ async def dataset_with_deletion(session):
             {"dataset_id": 1, "dataset_version_id": 2, "metadata_": {}},
         ],
     )
+
+
+@pytest.fixture
+async def datasets_with_interlaced_experiments(session):
+    dataset_ids = list(
+        await session.scalars(
+            insert(models.Dataset).returning(models.Dataset.id),
+            [{"name": f"{i}", "metadata_": {}} for i in range(3)],
+        )
+    )
+    dataset_version_ids = {
+        dataset_id: dataset_version_id
+        for dataset_id, dataset_version_id in await session.execute(
+            insert(models.DatasetVersion).returning(
+                models.DatasetVersion.dataset_id,
+                models.DatasetVersion.id,
+            ),
+            [{"dataset_id": dataset_id, "metadata_": {}} for dataset_id in dataset_ids],
+        )
+    }
+    await session.scalars(
+        insert(models.Experiment).returning(models.Experiment.id),
+        [
+            {
+                "dataset_id": dataset_id,
+                "dataset_version_id": dataset_version_ids[dataset_id],
+                "metadata_": {},
+            }
+            for _ in range(3)
+            for dataset_id in dataset_ids
```

Comment:
Nit: Maybe making `dataset_version_ids` a list and doing a `zip` is simpler.

Reply:
updated